pkg/kwokctl/components/etcd.go

```diff
@@ -73,9 +67,35 @@ func BuildEtcdComponent(conf BuildEtcdComponentConfig) (component internalversio
 		etcdArgs = append(etcdArgs,
 			"--data-dir=/etcd-data",
 		)
+
+		ports = append(
+			ports,
+			internalversion.Port{
+				HostPort: conf.Port,
```

Comment:
Not sure how to approach this one...
So if you start the cluster and not specify the port, we would expose this one, which is the default port 2379.
Should we default, or should we leave it to 0 in this case? This might cause problems on systems having etcd running as a service, if they do not specify the `--etcd-port`. I would lean more toward 0, but I can't really make a decision here :smile:

Reply:
Add a condition to this exposed port

``` golang
	exposePeerPort := true
	if conf.PeerPort == 0 {
		conf.PeerPort = 2380
		exposePeerPort = false
	}
	exposePort := true
	if conf.Port == 0 {
		conf.Port = 2379
		exposePort = false
	}

```

``` golang
	if exposePort {
		ports = append(
			ports,
			internalversion.Port{
				HostPort: conf.Port,
				Port:     2379,
			},
		)
	}
	if exposePeerPort {
		ports = append(
			ports,
			internalversion.Port{
				HostPort: conf.PeerPort,
				Port:     2380,
			},
		)
	}
```